main.go

```diff
@@ -72,8 +72,15 @@ func main() {
 	router.Use(gin.Logger())
 	router.Use(ginErrorLogMiddleware)
 
+	allowedOrigins := []string{"https://wallet.tx.xyz"}
+
+	if os.Getenv("CLIENT_ORIGIN") != "" {
+		// Place any specified origin at the front of the list
+		allowedOrigins = append([]string{os.Getenv("CLIENT_ORIGIN")}, allowedOrigins...)
+	}
+
 	router.Use(cors.New(cors.Config{
-		AllowOrigins:     []string{"http://localhost:3456", "https://wallet.tx.xyz"},
+		AllowOrigins:     allowedOrigins,
```

Comment:
I don't think we need wallet.tx.xyz as an allowed origin outside of prod. So if you want you can simplify to:
```
AllowOrigins: []string{os.Getenv("CLIENT_ORIGIN")}
```

And to make sure it's set in all envs you can check that this env var is set:
```
if os.Getenv("CLIENT_ORIGIN") == "" {
    log.Fatal("$CLIENT_ORIGIN must be set")
}
```

Reply:
i like it!